### ISSUE TITLE
Cloning Error

### ISSUE THREAD

**Issue by asharooni** [irsat000/CAI-Tools#91]
Hi! Thanks in advance for the extension.

Today I keep getting these errors when trying to duplicate different chats. I've attached an image of the error message!

![Screenshot 2024-12-01 022708](https://github.com/user-attachments/assets/de4e17e6-7ab6-443c-b5ae-913ea6abdf81)

**Comment by irsat000** [comment]
What about the result? Did it import or clone perfectly?

**Comment by asharooni** [comment]
No! It was bizarrely, partially cloned with shell messages sprinkled randomly throughout, unfortunately.

**Comment by irsat000** [comment]
The website is just too wonky, absolutely unpredictable and works different for everyone.

Not once in all my test it failed to deliver. So my eyes blind, ears deaf, trying to fix this.

**Comment by asharooni** [comment]
Yeah I understand! There's been so many base changes lately I can imagine something like c.ai tools on top of it would be confusing as all hell. If I end up getting any other hints as to what's going on my end I'll let ya know.

**Comment by irsat000** [comment]
Does this happen everytime? What's the rate of successful perfect chat clones?

**Comment by asharooni** [comment]
I used to have duplications no problem, this only began the day I posted here! At the moment, no matter what character, it errors. But before I could even duplicate way bigger chats no problem. It's hard to say what exactly goes wrong. Just now I tried with another chat and it did a completely different error, pictured here... it gave 'creating shells' the checkmark, then wouldn't start matching, going back to creating shells again before eventually deciding this error: 

![Screenshot 2024-12-04 233156](https://github.com/user-attachments/assets/ce5e50eb-5701-4c7d-b852-41ecffef0f91)

**Comment by irsat000** [comment]
You are making it very easy for me to erase this feature entirely. How is the old method by the way? Does it work with that? And what browser are you using?

**Comment by asharooni** [comment]
Oh that's too bad, I really value the feature! I haven't tried the old way, I didn't realize it still works despite the old website no longer running (probably misunderstood what it meant by 'old version'). I'll give that a try next! I'm on Chrome, currently version 131.0.6778.86 if that means anything haha. As a side note I just duplicated the same chat that errored earlier on the creating shells bit and it made it all the way to populating shells. The problem, same with the first error a few days back, seem to happen because  it keeps climbing higher and higher than what is expected. I've attached an image of it here. About to try via 'old version.'
![Screenshot 2024-12-04 234250](https://github.com/user-attachments/assets/151bea75-5b5c-460a-abb3-c84ac0f94e50)

**Comment by asharooni** [comment]
The old version led to some messages being right, and the majority being just 'Blank'. The literal word just... written there. And it isn't as if it reached a certain point and stopped working, it really is sporadic in accuracy.

**Comment by irsat000** [comment]
Old version was working fine for a long while, only errors were the ones interrupting it, not turning "blank". The website back end sucks.

**Comment by asharooni** [comment]
My guess is we shrug for now and let this function do its error-ridden thing?

**Comment by JCollins2048** [comment]
Hi, there.  I'm not experiencing this issue, but I might have an explanation.

The "can't create shells" error might have something to do with the current load on the server, or some kind of limiter on messages.  I know that *Janitor AI*, for example, only allows you to send one message every five seconds — which is only relevant if you've told the 'bot to say something like "(…)" for every reply. (I've manually imported chats this way.)

It's possible it may also have something to do with the user's computer being taxed during the import process.  I can't imagine opening a new chat and importing a bunch of messages would take a *lot* of processing power, even as fast as it does it, but I know that *Chrome* sometimes chugs under certain setups when CSS-heavy web pages are involved.

All that being said, one possible solution might be to slow down the rate at which the new chat is created and edited.

**Comment by irsat000** [comment]
What you are describing is called Rate Limiting. Chrome has no spam counter. CAI server is acyually too powerful and they use load balancing, it won't fail. Can't be CSS heavy, not relevant, because we directly open a WebSocket connection. Chrome only has parallel connection at a time limit, but I already made that accordingly.

**Comment by JCollins2048** [comment]
Well, it was just a thought.  It's reassuring that you already thought of the one relevant thing that I said, though, heh heh.

**Comment by irsat000** [comment]
I am thinking of a rework, combinding best of both worlds of old and new. In a few weeks or something.

**Comment by irsat000** [comment]
Have you tried Chatbot Manager's cloning?

**Comment by irsat000** [comment]
Could you try again with the new update? Then we can close this.

**Comment by asharooni** [comment]
Hi! Sorry to say I no longer use Character AI or your service, but hopefully the other person on this thread can check for you. Feel free to close the thread if not.

**Comment by irsat000** [comment]
What do you use by the way? I am getting ready to make tools for other chatbots.

**Comment by asharooni** [comment]
I make my own custom gpts on Chatgpt! The ability to have documents on my story canon etc for the AI to reference was a gamechanger. I've been wishing for a voice modifier for chatgpt (like the c.ai one so I can have a specific character's voice) and a way to get the regeneration button back for custom gpts (it's disabled when you use custom). Doubt chatgpt is that modifiable, but those are the dreams! 😭

**Comment by asharooni** [comment]
ps when I used c.ai your tools were a gamechanger, so thanks again.

**Comment by irsat000** [comment]
Such coincidence. I was planning on working on Chatgpt.

You can always write your ideas here.

**Comment by asharooni** [comment]
Ugh, divine intervention to be sure 🤣 I will definitely post here if I get any other ideas. Please keep me updated! There's a helpful extension I've been using "GPT Reader: A free ChatGPT powered TTS" lately, and if you ever manage to make it so I can make and use my own voices somehow... Damn. Anyways. Excited for whatever you think up, regardless of what I'm hoping for.
